src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const string = `${separator}`;
+
+    for (const word of this) {
+      if (word === null || word === undefined) {
+        result += string;
```

Comment:
you need to add a separator every time except for the last element, think about that